lib/miq_expression/remove_equality_visitor.rb

```diff
@@ -0,0 +1,71 @@
+class MiqExpression
+  # rubocop:disable Naming/MethodName
+  class RemoveEqualityVisitor < Arel::Visitors::Reduce
+    def visit_NilClass(object, _collector)
+      object
+    end
+
+    def visit_Arel_Nodes_Equality(object, collector)
+      field, binds = collector
```

Comment:
Just to clarify:  `binds` here is getting mutated if `object.right` is a `Arel::Nodes::BindParam`, correct?  Same with `object` in that case as well?

Also, could we change `field` to `node_to_remove` here as well?  Would make it a bit more obvious what is being done here, and since `field` isn't part of the original method signature, I think that is fine (`object` on the other hand is and should probably stay as-is).